pkg/userdata/upgrade_test.go

```diff
@@ -0,0 +1,184 @@
+      networking:
+        dnsDomain: cluster.local
+        podSubnet: 10.244.0.1/16
+        serviceSubnet: 10.96.0.1/12
+      ---
+      apiVersion: kubeproxy.config.k8s.io/v1alpha1
+      kind: KubeProxyConfiguration
+      mode: ipvs
+      ipvs:
+        scheduler: lc
+  trustd:
+    username: 'test'
+    password: 'test'
+    endpoints: [ 1.2.3.4, 2.3.4.5, 3.4.5.6 ]
+    certSANs: [ 1.2.3.4 ]
+
+networking:
+  os:
+    devices:
+    - interface: lo
+      cidr: 1.2.3.4/32
+    - interface: eth0
+      dhcp: true
+install:
+  wipe: false
+  force: true
+  boot:
+    device: /dev/sda
+    size: 1024000000
+    kernel: http://127.0.0.1:8080/assets/talos/version/vmlinuz
+    initramfs: http://127.0.0.1:8080/assets/talos/version/initramfs.xz
+  root:
+    device: /dev/sda
+    size: 2048000000
+    rootfs: http://127.0.0.1:8080/assets/talos/version/rootfs.tar.gz
+  data:
+    device: /dev/sda
+    size: 4096000000
+`
+
+const joinConfig = `version: ""
+security:
+  os:
+    ca:
+      crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0=
+      key: LS0tLS1CRUdJTiBFQyBQUklWQVRFIEtFWS0tLS0tCi0tLS0tRU5EIEVDIFBSSVZBVEUgS0VZLS0tLS0=
+    identity:
+      crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0=
+      key: LS0tLS1CRUdJTiBFQyBQUklWQVRFIEtFWS0tLS0tCi0tLS0tRU5EIEVDIFBSSVZBVEUgS0VZLS0tLS0=
+  kubernetes:
+    ca:
+      crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0=
+      key: LS0tLS1CRUdJTiBFQyBQUklWQVRFIEtFWS0tLS0tCi0tLS0tRU5EIEVDIFBSSVZBVEUgS0VZLS0tLS0=
+networking:
+  os:
+    devices:
+    - interface: lo
+      cidr: 1.2.3.4/32
+    - interface: eth0
+      dhcp: true
+services:
+  init:
+    cni: flannel
+  kubeadm:
+    configuration: |
+      apiVersion: kubeadm.k8s.io/v1beta1
+      kind: JoinConfiguration
+      controlPlane: {}
+      discovery:
+        bootstrapToken:
+          token: '123456.1234567890987654'
+          unsafeSkipCAVerification: true
+          apiServerEndpoint: 2.3.4.5:443
+      nodeRegistration:
+        taints:
+        kubeletExtraArgs:
+          node-labels:
+          feature-gates: ExperimentalCriticalPodAnnotation=true,TTLAfterFinished=true
+  trustd:
+    username: 'test'
+    password: 'test'
+    endpoints: [ 1.2.3.4, 2.3.4.5, 3.4.5.6 ]
+    certSANs: [ 1.2.3.4 ]
+install:
+  boot:
+    device: /dev/sda
+    size: 1024000000
+    kernel: http://127.0.0.1:8080/assets/talos/version/vmlinuz
+    initramfs: http://127.0.0.1:8080/assets/talos/version/initramfs.xz
+  root:
+    device: /dev/sda
+    size: 2048000000
+    rootfs: http://127.0.0.1:8080/assets/talos/version/rootfs.tar.gz
+  data:
+    device: /dev/sda
+    size: 4096000000
+  wipe: false
+  force: true
+`
+
+type upgradeSuite struct {
+	suite.Suite
+}
+
+func TestUpgradeSuite(t *testing.T) {
+	suite.Run(t, new(upgradeSuite))
+}
+
+// func (suite *upgradeSuite) TestUpgrade() {
```

Comment:
This intentional?

Reply:
Yeah I can't get the test to work due to the way kubeadm marshaling works :/